src/CloudNative.CloudEvents/CloudEvent.cs

```diff
@@ -390,10 +390,11 @@ public CloudEvent Validate()
         /// <summary>
         /// Validates that this CloudEvent is valid in the same way as <see cref="IsValid"/>,
         /// but throwing an <see cref="ArgumentException"/> using the given parameter name
-        /// if the event is invalid. This is typically used within protocol bindings as the last
-        /// step in conversion.
+        /// if the event is invalid. This is typically used within protocol bindings or event formatters
+        /// as the last step in decoding an event, or as the first step when encoding an event.
         /// </summary>
-        /// <param name="paramName">The parameter name to use in the exception if the event is invalid.</param>
+        /// <param name="paramName">The parameter name to use in the exception if the event is invalid.
```

Comment:
Why is the parameter name important? Doesn't the error message include enough information since it lists the missing required attributes?

Reply:
I just generally think it's a good idea to provide a parameter name in `ArgumentException`, so callers know which part of their call is invalid.

We *could* potentially drop that... it's only a nice-to-have.